### ISSUE TITLE
video.min.js errors in IE8, must use video.js

### ISSUE THREAD

**Issue by sethborg** [videojs/video.js#3064]
Something's busted about video.min.js since it breaks IE8 support. Could someone look into this or if you aren't going to fix it, at least update the docs with an html comment in the source code so people know

Use the below to see the error, change video.min.js to video.js and see how it works fine

&lt;!DOCTYPE html&gt;
&lt;html lang="en"&gt;
&lt;head&gt;
	&lt;link href="http://vjs.zencdn.net/5.5.3/video-js.css" rel="stylesheet"&gt;
	&lt;script src="http://vjs.zencdn.net/ie8/1.1.1/videojs-ie8.min.js" &gt;&lt;/script&gt;
	&lt;!-- must use video.js, IE8 errors if you use video.min.js --&gt;
	&lt;script src="http://vjs.zencdn.net/5.5.3/video.min.js" &gt;&lt;/script&gt;
&lt;/head&gt;

&lt;body&gt;
  &lt;video controls="controls" class="video-js vjs-fluid vjs-default-skin vjs-big-play-centered" preload="auto" data-setup='{}'&gt;
    &lt;source src="http://clips.vorwaerts-gmbh.de/VfE_html5.mp4" type="video/mp4"&gt;
    &lt;p class="vjs-no-js"&gt;
      To view this video please enable JavaScript, and consider upgrading to a web browser that
      &lt;a href="http://videojs.com/html5-video-support/" target="_blank"&gt;supports HTML5 video&lt;/a&gt;
    &lt;/p&gt;
  &lt;/video&gt;
&lt;/body&gt;
&lt;/html&gt;

**Comment by gkatsev** [comment]
Can you elaborate on what you mean by ie8 being broken? Are there any errors in the console? What behavior are you seeing and what behavior are you expecting to see?
Thanks.

**Comment by sethborg** [comment]
Yeah console error and the Flash video player never loads because of it

**Comment by gkatsev** [comment]
What error are you seeing?

**Comment by sethborg** [comment]
Nothing useful, that's why I didn't include it:

Script error
video.min.js
Code: 0
URI: http://vjs.zencdn.net/5.5.3/video.min.js
Line: 0
Char: 0

**Comment by gkatsev** [comment]
Can't know unless I ask. A lot of times errors do contain useful information but the people reporting the issue don't think it's useful so they don't provide it. It's generally best to provide such information even if at first glance it seems not useful.
Thanks, we'll investigate.

**Comment by kingprimex** [comment]
Have you tried to use compatibility mode set to IE9? For example, JWPlayer 7 flash source not working on IE8 or lower! I suspect the same behaviour with VideoJS
![videojs](https://cloud.githubusercontent.com/assets/6496606/12765856/43e62e42-ca09-11e5-91c7-c462d10f8c1d.JPG)

**Comment by QuietJam** [comment]
I've also met this problem. It's probably because of some bug which is made when compressing code after using ES6 syntax and compiling it to ES5.

**Comment by sethborg** [comment]
I tried adding &gt;meta http-equiv="X-UA-Compatible" content="IE=Edge" /&lt; to the page but it didn't change anything. Probably because that applies to the DOM not javascript.

There's some issue with your minimizing function that makes it incompatible with IE8.

**Comment by sethborg** [comment]
I just tested this in IE8 on 5.10.0 and this bug still exists. See also https://github.com/videojs/video.js/issues/3142

What I've found is that I must use video.js not video.min.js and I must use: videojs('video', {}, function(){ }); instead of data-setup="{}" because it errors on line 13611 which I've marked below.

`  // Check if any media elements exist
  if (mediaEls && mediaEls.length > 0) {

    for (var i = 0, e = mediaEls.length; i < e; i++) {
      var mediaEl = mediaEls[i];

      // Check if element exists, has getAttribute func.
      // IE seems to consider typeof el.getAttribute == 'object' instead of 'function' like expected, at least when loading the player immediately.
      if (mediaEl && mediaEl.getAttribute) {

        // Make sure this player hasn't already been set up.
        if (mediaEl['player'] === undefined) {
          var options = mediaEl.getAttribute('data-setup');

          // Check if data-setup attr exists.
          // We only auto-setup if they've added the data-setup attr.
          if (options !== null) {
            // Create new video.js instance.
            var player = videojs(mediaEl);     <- IT ERRORS HERE
          }
        }

        // If getAttribute isn't defined, we need to wait for the DOM.
      } else {
          autoSetupTimeout(1);
          break;
        }
    }

    // No videos were found, so keep looping unless page is finished loading.
  } else if (!_windowLoaded) {
      autoSetupTimeout(1);
    }`

**Comment by gkatsev** [comment]
Have a reduced test case? When I was testing the fix (https://github.com/videojs/video.js/pull/3233), it was working fine for me in IE8 as well.

**Comment by sethborg** [comment]
reduced test cases

http://ivcmiami.com/testvideojs/min_js_setup.htm   <- broken
http://ivcmiami.com/testvideojs/min_data_setup.htm   <- works
http://ivcmiami.com/testvideojs/no_min_js_setup.htm   <- works
http://ivcmiami.com/testvideojs/no_min_data_setup.htm   <- works

**Comment by gkatsev** [comment]
@sethborg looks like the minified version is erroring out when you pass in a player id as a string to it. This is actually occurring because the minifier is doing weird stuff that is causing issues.
specially, it turns the main `videojs` function from (simplified):
```js
var videojs = function(id, options, ready) {
   return videojs.getPlayers()[id];
```
to something like
```js
var videojs = function vjs(id, options, ready) {
  return vjs.getPlayers()[id];
```
However, in IE8, `vjs` and `videojs` are *not* equal inside the function and thus it doesn't have the propery `getPlayers` that we expect.
Reopening and investigating why our minifier produces this code since it is also technically 2 characters longer than necessary.

**Comment by gkatsev** [comment]
Also, interestingly enough, if you pass in an element reference instead of a string id, it works just fine. I guess because it doesn't enter the main block that does `videojs.getPlayers()`

**Comment by gkatsev** [comment]
Looks like the issue isn't uglify but rather that babel produces a named function expression that then uglify changes the names of.

**Comment by gkatsev** [comment]
Fixed in 5.10.5.
